### DISCUSSION TITLE
Multiple input files

### DISCUSSION THREAD

**Discussion by liangjs** [foonathan/lexy#143] [Q&A] [answered]
How do we handle multiple input files during parsing? For example, when the C/C++ preprocessor encounters the `include` macro, it needs to continue parsing the included file.

**Comment by liangjs** [comment]
By the way, how to get the current file name when reporting errors?

**Reply by foonathan** [reply]
You need to store the file name in the error reporting callback. If you're using `lexy_ext::report_error` you can use `.path`: https://github.com/foonathan/lexy/blob/b53bd4a8fe472aaec89ad50ff805797980e6fbb6/include/lexy_ext/report_error.hpp#L360-L364

**Comment by foonathan** [comment, marked as answer]
This isn't something lexy supports natively. You need to record the `#include` statement and resolve it in a second pass.

If you absolutely need the contents to continue parsing, which is an unfortunate situation to be in, you can do that directly in side a scan rule: https://lexy.foonathan.net/reference/dsl/scan/#scan
